public/app/plugins/datasource/grafana-azure-monitor-datasource/components/InsightsConfig.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { PureComponent } from 'react';
+import { FormLabel, Button, Input } from '@grafana/ui';
+
+export interface Props {
+  datasourceConfig: any;
+  onDatasourceUpdate: (config: any) => void;
+}
+
+export interface State {
+  config: any;
+}
+
+export class InsightsConfig extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const { datasourceConfig } = this.props;
+
+    this.state = {
+      config: datasourceConfig,
+    };
+  }
+
+  static getDerivedStateFromProps(props: Props, state: State) {
+    return {
+      ...state,
+      config: props.datasourceConfig,
+    };
+  }
+
+  onAppInsightsAppIdChange = (appInsightsAppId: string) => {
+    this.props.onDatasourceUpdate({
+      ...this.state.config,
+      editorJsonData: {
+        ...this.state.config.editorJsonData,
+        appInsightsAppId,
+      },
+    });
+  };
+
+  onAppInsightsApiKeyChange = (appInsightsApiKey: string) => {
+    this.props.onDatasourceUpdate({
+      ...this.state.config,
+      editorSecureJsonData: {
+        ...this.state.config.editorSecureJsonData,
+        appInsightsApiKey,
+      },
+    });
+  };
+
+  onAppInsightsResetApiKey = () => {
+    this.props.onDatasourceUpdate({
+      ...this.state.config,
+      version: this.state.config.version + 1,
+      secureJsonFields: {
+        ...this.state.config.secureJsonFields,
+        appInsightsApiKey: false,
+      },
+    });
+  };
+
+  render() {
+    const { config } = this.state;
+    return (
+      <>
+        <h3 className="page-heading">Application Insights Details</h3>
+        <div className="gf-form-group">
+          {config.secureJsonFields.appInsightsApiKey ? (
```

Comment:
Multiple Applications needs to be supported, we would show a list of applications/apikey pairs, with ability to add/delete.

![image](https://user-images.githubusercontent.com/7364245/62406509-8bed6180-b572-11e9-89bd-ea9a3fe84d05.png)

Reply:
This seems like something that should be another issue. This PR is already pretty big.

Reply:
Agree with Shavonn here - this can be a new PR.